conan/tools/files/files.py

```diff
@@ -264,17 +265,53 @@ def chdir(conanfile, newdir):
         os.chdir(old_path)
 
 
-def chmod(conanfile, path:str, read:bool=None, write:bool=None, execute:bool=None, recursive:bool=False):
-    """
-    Change the permissions of a file or directory. The same as the Unix command chmod, but simplified.
+def chmod(conanfile, path:str, read:Optional[bool]=None, write:Optional[bool]=None, execute:Optional[bool]=None, recursive:Optional[bool]=False):
```

Comment:
Why is recursive `Optional`? Can it be `None`?

Reply:
No, it only can be True/False. Fixed by https://github.com/conan-io/conan/pull/18038/commits/3511a9b7fbf1ad24b65849cd6254de2bd99d81cf